halo/attest/keeper/valaddr.go

```diff
@@ -59,10 +65,10 @@ func (c *valAddrCache) SetAll(vals []*vtypes.Validator) error {
 // It uses the validator read-through-cache to avoid querying the underlying validator set provider.
 // Note it assumes the provided validator is inside the current set. It doesn't ensure this.
 func (k *Keeper) getValEthAddr(ctx context.Context, cmtAddr []byte) (common.Address, error) {
-	if len(cmtAddr) != crypto.AddressSize {
-		return common.Address{}, errors.New("invalid comet address length [BUG]", "len", len(cmtAddr))
+	addr, err := cast.Array20(cmtAddr)
+	if err != nil {
+		return common.Address{}, errors.Wrap(err, "invalid comet address length [BUG]", "len", len(cmtAddr))
```

Comment:
wouldn't the length already be reported by the error from cast

Reply:
yup